docs/DeveloperGuide.md

```diff
@@ -6,8 +6,70 @@
 
 ## Design & implementation
 
+### Architecture
+
+### UI Component
+
+# UserCommand feature
+
+User can type command to do things on the Map. 
+
+If the user type the command, `parseCommand` function will be called. After that, the original input will be
+analyzed by `analyseCommand` function to see if it matches a kind of `CommandType`. Then, we will return new
+Command back to the main based on the command type. The Final step is to call the `execute` function. If the command
+is the type of `fightCommand`, we will call the execute function with one parameter `Scanner`. For all other
+conditions, we will call the execute function with no parameter.
+
+![dummy test](/picture/Command.svg)
+
```

Comment:
Perhaps it could be helpful to provide a few more examples of inputs and outputs for correct program behaviour